core/src/main/java/org/apache/accumulo/core/client/admin/servers/Server.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.accumulo.core.client.admin.servers;
+
+import java.util.Comparator;
+import java.util.Objects;
+
+import com.google.common.base.Preconditions;
+
+public abstract class Server implements Comparator<Server>, Comparable<Server> {
```

Comment:
Wonder if this should have a more specific name.  Could be ServerAddress or AccumuloServerAddress.  If the group info were to be included in this class, then maybe AccumuloServer or AccumuloServerInfo

Reply:
This seems more like a ServerId rather than a Server object. It contains identifying information about a server, which may or may not exist.

This could extend AbstractId, and be constructed similarly to those, to avoid creating redundant instances that refer to the same server.

This should also not implement `Comparator<Server>`, it should be something like `Server<T extends Server<?>> implements Comparable<T>`, so that 1) it's a Comparable object, not a Comparator, and 2) it is only comparable to instances of the same type.